### PR TITLE
DAOS-7821 gha: Change github actions cache names and improve coverage.

### PR DESCRIPTION
**Pull request by ashleypittman**
Update github actions cache names to improve re-use.
Update the build to run NLT testing on more distros.
Fix SPDK build to make binaries that can run across different CPUs.

The github actions hash key was intended to be <update-of-build-scripts>-<HEAD>
however as landing builds were not checked out with commit history
enabled they were <HEAD>-<HEAD> which worked, however meant that
PRs which used older versions of build scripts would get most recent master
build rather than most recent master where the change was made, so the
overhead for these builds was higher.

Speed up master builds slightly, we slowed them right down before to not
complete with PRs but now GHA is often idle other than throttled master
builds.

Run NLT across all distros and fix missing packages to do do.  Enable NLT
on all distros for master builds, assuming the SPDK change is stable then
we can enable this for PRs later.

### CODE REVIEW

File: site_scons/env_modules.py [hunk 1]
```diff
@@ -39,14 +39,10 @@ def __init__(self):
...
-                                                          "modules"),
-                                             os.path.sep +
-                                             os.path.join("etc",
-                                                          "modulefiles")] +
-                                            os.environ.get("MODULEPATH",
-                                                           "").split(":"))
+        os.environ["MODULEPATH"] = ":".join([os.path.join("/usr", "share", "modules"),
+                                             os.path.join("/usr", "share", "modulefiles"),
+                                             os.path.join("/etc", "modulefiles")] +
+                                            os.environ.get("MODULEPATH", "").split(":"))
```
Comment (knard-intel) [inline site_scons/env_modules.py:45]: nitpicking remark
```suggestion
                                             os.environ.get("MODULEPATH", "").split(":"))
```

File: site_scons/env_modules.py [hunk 2]
```diff
@@ -39,14 +39,10 @@ def __init__(self):
...
         # in it's MODULEPATH, for some unknown reason
-        os.environ["MODULEPATH"] = ":".join([os.path.sep +
-                                             os.path.join("usr", "share",
-                                                          "modules"),
-                                             os.path.sep +
-                                             os.path.join("etc",
-                                                          "modulefiles")] +
-                                            os.environ.get("MODULEPATH",
-                                                           "").split(":"))
+        os.environ["MODULEPATH"] = ":".join([os.path.join("/usr", "share", "modules"),
```
Comment (knard-intel) [inline site_scons/env_modules.py:42]: nitpicking
```suggestion
        os.environ["MODULEPATH"] = ":".join([os.path.join(os.sep, "usr", "share", "modules"),
```

Comment (ashleypittman) [reply site_scons/env_modules.py:42]: Oh yes, that's probably better.

File: utils/node_local_test.py [hunk 3]
```diff
@@ -671,7 +671,8 @@ def start(self):
         cmd = ['storage', 'format', '--json']
         while True:
             try:
-                self._sp.wait(timeout=0.5)
+                rc = self._sp.wait(timeout=0.5)
+                print(rc)
```
Comment (knard-intel) [inline utils/node_local_test.py:675]: Maybe a more explicit message could be helpful for the end user.
I am also supposing that this is some old debugging traces which have not yet been removed?

File: src/include/gurt/common.h [hunk 4]
```diff
@@ -162,24 +162,15 @@ char *d_realpath(const char *path, char *resolved_path);
...
-		if (errno == 0 || errno == ENOMEM) {			\
-			int _size = 0;					\
-			void *_ptr = (ptr);				\
-			if (_ptr)					\
-				_size = strnlen(_ptr, PATH_MAX + 1) + 1 ; \
+		if ((ptr) != NULL) {					\
+			int _size = strnlen(ptr, PATH_MAX + 1) + 1 ;	\
 			D_CHECK_ALLOC(realpath, true, ptr, #ptr, _size,	0, #ptr, 0); \
-			if (errno == 0 && ((ptr) == NULL) && _ptr != NULL) \
+			if (((ptr) == NULL))				\
```
Comment (jolivier23) [inline src/include/gurt/common.h:172]: Is this change intentional?   I don't see it documented and it seems like a regression from your prior fix

**Review by jolivier23 — CHANGES_REQUESTED** [review]

File: src/include/gurt/common.h (hunk 4 above)
Comment (ashleypittman) [reply src/include/gurt/common.h:172]: Yes, on Leap then realpath will succeed yet set errno to EINVAL so this change is required on that distro.

File: utils/node_local_test.py (hunk 3 above)
Comment (knard-intel) [reply utils/node_local_test.py:675]: @ashleypittman , please could you tell me (just for my understanding), if this print is used for debugging purpose or if I miss something.
Thanks in advance.

Comment (ashleypittman) [reply utils/node_local_test.py:675]: It's debugging.  What it prints is the exit code of daos_server if it's exited, if this is -4 then we can tell that the exit was because of SIGILL which is the spdk build issue.

**Review by tanabarr — APPROVED** [review]
somewhat unfamiliar domain but looks good from what I can see

**Review by jolivier23 — APPROVED** [review]
